src/toolkit/matl_sell_policy.cc

```diff
@@ -208,22 +239,36 @@ void MatlSellPolicy::GetMatlTrades(
     std::vector<std::pair<Trade<Material>, Material::Ptr> >& responses) {
   Composition::Ptr c;
   std::vector<Trade<Material> >::const_iterator it;
+
+  // confirm that trades are within transport unit limits
+  int shippable_pkgs = transport_unit_->MaxShippablePackages(trades.size());
+
+  double qty;
+
   for (it = trades.begin(); it != trades.end(); ++it) {
-    double qty = it->amt;
-    LGH(INFO3) << " sending " << qty << " kg of " << it->request->commodity();
-    Material::Ptr mat = buf_->Pop(qty, cyclus::eps_rsrc());
-    Material::Ptr trade_mat;
+    if (shippable_pkgs > 0){
```

Comment:
The indentation is a little off making it hard to read, but...  it seems like this if statement could be outside the loop?

Reply:
Sorry, the indentation below in the loop was wrong. it should be checking at each trade whether there are stlll `shippable_pkgs` left to ship, and everything below happens only if there are still `shippable_pkgs`